src/stdio_mgr/stdio_mgr.py

```diff
@@ -27,14 +27,55 @@
 """
 
 import sys
-from contextlib import contextmanager
-from io import BufferedReader, BytesIO, StringIO, TextIOBase, TextIOWrapper
+from contextlib import contextmanager, ExitStack, suppress
+from io import BufferedRandom, BufferedReader, BytesIO, TextIOBase, TextIOWrapper
 
 import attr
 
 
+class _SafeCloseIOBase(TextIOBase):
+    """Class to ignore ValueError during close.
+
+    Subclass of :cls:`~io.TextIOBase` that disregards ValueError which can
+    occur if the file has already been closed.
+    """
+
+    def close(self):
+        """Close and suppress ValueError.
+
+        ValueError may occur when the underlying
+        buffer is detached or the file was closed.
+        """
+        with suppress(ValueError):
+            super().close()
+
+
+class RandomTextIO(_SafeCloseIOBase, TextIOWrapper):
+    """Class to capture writes to a buffer even when detached.
```

Comment:
Maybe add a blurb about `.getvalue()` to this docstring, too? Perhaps even just duplicate [this](https://github.com/bskinn/stdio-mgr/pull/26/files#diff-abadc7d46c647e14e338ebfbc175cc05R91-R93)?

Reply:
ok. im not used to describing the methods in the cls doc

Reply:
In a case like this, where methods not present in a parent class are *added*, as opposed to method already present being *overridden*, I'm inclined to put some narrative text in the class docstring to make the additions more noticeable to the reader.  The method docstring can then be very clinical/technical/terse.